scripts/src/javatests/org/oppia/android/scripts/common/MavenDependenciesListGeneratorTest.kt

```diff
@@ -0,0 +1,1330 @@
+import com.google.protobuf.TextFormat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.oppia.android.scripts.maven.data.MavenListDependency
+import org.oppia.android.scripts.proto.DirectLinkOnly
+import org.oppia.android.scripts.proto.ExtractedCopyLink
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.ScrapableLink
+import org.oppia.android.scripts.testing.TestBazelWorkspace
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.util.concurrent.TimeUnit
+
+/** Tests for [MavenDependenciesListGenerator]. */
+class MavenDependenciesListGeneratorTest {
+
+  private val THIRD_PARTY_PREFIX = "//third_pary:"
+  private val DEP_WITH_SCRAPABLE_LICENSE = "androidx.databinding:databinding-adapters:3.4.2"
+  private val DEP_WITH_NO_LICENSE = "com.google.protobuf:protobuf-lite:3.0.0"
+  private val DEP_WITH_SCRAPABLE_AND_EXTRACTED_COPY_LICENSES =
+    "com.github.bumptech.glide:annotations:4.11.0"
+  private val DEP_WITH_DIRECT_LINK_ONLY_LICENSE = "com.google.firebase:firebase-analytics:17.5.0"
+  private val DEP_WITH_INVALID_LINKS = "io.fabric.sdk.android:fabric:1.4.7"
+  private val DEP_WITH_SAME_SCRAPABLE_LICENSE_BUT_DIFFERENT_NAME =
+    "com.squareup.moshi:moshi:1.11.0"
+
+  private val DATA_BINDING_VERSION = "3.4.2"
+  private val PROTO_LITE_VERSION = "3.0.0"
+  private val GLIDE_ANNOTATIONS_VERSION = "4.11.0"
+  private val FIREBASE_ANALYTICS_VERSION = "17.5.0"
+  private val IO_FABRIC_VERSION = "1.4.7"
+  private val MOSHI_VERSION = "1.11.0"
+
+  private val DATA_BINDING_POM = "https://maven.google.com/androidx/databinding/databinding-" +
+    "adapters/$DATA_BINDING_VERSION/databinding-adapters-$DATA_BINDING_VERSION.pom"
+  private val PROTO_LITE_POM = "https://repo1.maven.org/maven2/com/google/protobuf/protobuf" +
+    "-lite/$PROTO_LITE_VERSION/protobuf-lite-$PROTO_LITE_VERSION.pom"
+  private val IO_FABRIC_POM = "https://maven.google.com/io/fabric/sdk/android/fabric/" +
+    "$IO_FABRIC_VERSION/fabric-$IO_FABRIC_VERSION.pom"
+  private val GLIDE_ANNOTATIONS_POM = "https://repo1.maven.org/maven2/com/github/bumptech/glide" +
+    "/annotations/$GLIDE_ANNOTATIONS_VERSION/annotations-$GLIDE_ANNOTATIONS_VERSION.pom"
+  private val FIREBASE_ANALYTICS_POM = "https://maven.google.com/com/google/firebase/firebase-" +
+    "analytics/$FIREBASE_ANALYTICS_VERSION/firebase-analytics-$FIREBASE_ANALYTICS_VERSION.pom"
+  private val MOSHI_POM = "https://repo1.maven.org/maven2/com/squareup/moshi/moshi/" +
+    "$MOSHI_VERSION/moshi-$MOSHI_VERSION.pom"
+
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "Licenses details are not completed"
```

Comment:
Is this actually needed in this suite? Prefer removing all vals not used by these tests.

Reply:
Removed.